src/main/java/lotto/domain/LottoGroup.java

```diff
@@ -0,0 +1,60 @@
+package lotto.domain;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class LottoGroup {
+    private final List<Lotto> lottos;
+
+    public LottoGroup(Money money) {
+        this(buyLotto(money));
+    }
+
+    public LottoGroup(List<Lotto> lottos) {
+        this.lottos = lottos;
+    }
+    private static List<Lotto> buyLotto(Money money) {
+        int cnt = money.getBuyableCount();
+
+        List<Lotto> lottoArray = new ArrayList<>();
+
+        for(int i = 0; i < cnt; i++) {
+            lottoArray.add(Lotto.of(LottoMachine.createLottoNumbers()));
+        }
+
+        return lottoArray;
+    }
+
+    public List<Lotto> getLottoNumbers() {
+        return lottos;
+    }
+
+    public LottoResult match(Lotto winLotto) {
+        return new LottoResult(calculate(winLotto));
+    }
+
+    private  Map<LottoRank, Integer> calculate(Lotto winLotto) {
+        Map<LottoRank, Integer> reusltMap = initResult();
+
+        for (Lotto lotto : this.lottos) {
+            int cnt = lotto.matchedCount(winLotto);
+            LottoRank lottoRank = LottoRank.getLottoRank(cnt);
+
+            if (lottoRank != null) {
+                reusltMap.put(lottoRank, reusltMap.get(lottoRank) + 1);
+            }
+        }
+
+        return reusltMap;
```

Comment:
```suggestion
    LottoResult result = new LottoResult();
    for (Lotto lotto : lottos) {
      Rank rank = lotto.match(winLotto);
      result.rank(rank);
    }
    return result;
```
현재 구현하고 있는 로직은 대표적인 절차지향적인 사고로 구현한 로직임
위와 같이 Map에 대한 상태 변경을 Map을 상태 값으로 가지고 있는 객체가 담당하도록 상태 값을 가지는 도메인 객체를 좀 더 능동적인 객체로 바라보기 위해 노력해 본다.
이렇게 구현했을 때 테스트 관점에서 얻을 수 있는 잇점은?